06.bowling_object/frame.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+require './shot'
+
+class Frame
+  attr_reader :shots
+
+  def initialize(shots)
+    @shots = shots.map { |shot| Shot.new(shot) }
+  end
+
+  def score(scores, frame_count)
+    frame_score = sum_frame(@shots)
+    return frame_score if frame_score != 10 || frame_count == 9
+
+    next_frame = scores[frame_count + 1]
+    frame_score + if @shots[0].strike? && frame_count == 8
+                    sum_frame(next_frame.shots[0, 2])
+                  elsif @shots[0].strike?
+                    strike(next_frame.shots, scores[frame_count + 2].shots)
+                  else
+                    spare(next_frame.shots)
+                  end
+  end
+
+  private
+
+  def sum_frame(shots)
+    shots.map(&:point).sum
+  end
+
+  def strike(next_shots, after_next_shots)
+    next_shots[0].strike? ? 10 + after_next_shots[0].point : sum_frame(next_shots)
```

Comment:
```suggestion
  def strike(next_frame, after_next_frame)
    next_frame.strike? ? 10 + after_next_frame.shots[0].point : sum_frame(next_frame.shots)
```

Frame自身が「自分がストライクかどうか」を判定できるようになってるといいですね

あと、メソッド名のstrikeというのは処理内容を正しく表現できていない気がします。
https://qiita.com/jnchito/items/459d58ba652bf4763820

Reply:
Frame自身にインスタンス変数のis_strikeを持たせて判定できるようにし、calc_strike,calc_spareに変更しました。